### ISSUE TITLE
misaligned stack access when using kstack

### ISSUE THREAD

**Issue by yump** [bpftrace/bpftrace#3644]
### Minimal reproduction

```
> sudo bpftrace -ve 'profile:hz:1 { @[pid,kstack]=count(); }'
node count: 8
Attaching 1 probe...
load profile:hz:1, with BTF, with func_infos = -13: Permission denied
load profile:hz:1, with BTF = -13: Permission denied
load profile:hz:1, version: 6.11.10 = -13: Permission denied
load profile:hz:1, version: 6.11.10 = -13: Permission denied
load profile:hz:1, version: 6.11.3 = -13: Permission denied

Error log: 
0: R1=ctx() R10=fp0
0: (bf) r7 = r1                       ; R1=ctx() R7_w=ctx()
1: (85) call bpf_get_current_pid_tgid#14      ; R0_w=scalar()
2: (77) r0 >>= 32                     ; R0_w=scalar(smin=0,smax=umax=0xffffffff,var_off=(0x0; 0xffffffff))
3: (7b) *(u64 *)(r10 -44) = r0
misaligned stack access off 0+0+-44 size 8
processed 4 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0

ERROR: Error loading program: profile:hz:1
```

Using binaries from the releases page, this problem first appeared in 0.21.0. Version 0.20.4 is unaffected.

### `bpftrace --info` output

```
System
  OS: Linux 6.11.10-300.fc41.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Nov 23 00:51:20 UTC 2024
  Arch: x86_64

Build
  version: v0.21.2
  LLVM: 19.1.0
  unsafe probe: no
  bfd: yes
  liblldb (DWARF support): yes
  libsystemd (systemd notify support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: no
  get_boot_ns: yes
  dpath: yes
  skboutput: yes
  get_tai_ns: yes
  get_func_ip: yes
  jiffies64: yes
  for_each_map_elem: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: yes
  module btf: yes
  map batch: yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes
  ringbuf: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
  kprobe_multi: yes
  uprobe_multi: yes
  raw_tp_special: yes
  iter: yes
```

**Comment by jordalgo** [comment]
Interesting. I wasn't able to repro this on master with llvm 19.1.4. Are you able to try with a local build off master?

**Comment by jordalgo** [comment]
Ok, I was able to repro with the 0.21.2 release on a 6.11 and a 5.19 kernel.

**Comment by jordalgo** [comment]
Same issue `BEGIN { @[1, "a"]=count(); }`

**Comment by jordalgo** [comment]
@yump This is now fixed in version 0.21.3
